src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -434,5 +434,15 @@ public static class Diagnostics {
 			isEnabledByDefault: true,
 			description: "The parameter {0} has a default value of {1} here, but {2} in its original definition in {3}. This causes inconsistent behaviour. Please use the same defualt value everywhere."
 		);
+
+		public static readonly DiagnosticDescriptor BannedConfig = new DiagnosticDescriptor(
+			id: "D2LXXXX",
+			title: "Fetching the config variable \"{0}\" has been banned. {1}",

```

Comment:
I wanted a "nicer" word than banned, but couldn't think of one, outside of deprecated? I'll just do deprecated